### ISSUE TITLE
Add Documentation

### ISSUE THREAD

**Issue by oubiwann** [clojusc/ring-xml#24]
Features:
 * [x] Improved README - #25
 * [x] Add Codox support - #26
 * [ ] Add Guides - #27

Part of release #3

**Comment by oubiwann** [comment]
@ckampfe, that'd be super-cool if you had the time :-) 

As for your docs question, two things:
 * Generated API reference (Codox will do the right thing ... as long as the docstrings are there!)
 * Examples/Guides

I suspect your questions was targeted at the latter. I'm thinking about doing the same thing here as I did with the friend-oauth2 project: put the examples in `./examples/` and have the `project.clj` file set that as a source for an `examples` profile (and probably `test` profile, too). This will be code that I regularly run before a release (in addition to the unit tests) to ensure that apps using `ring-xml` still work with whatever new features have landed. You know, manual integration tests :-)

With the examples created, user guides will be pretty straight-forward: "This is a `ring--xml` app that demonstrates X feature in the library. Here's how you add it to your app, here's how you get the data, etc."

Basically, mini-walkthroughs of the mini-examples :-)

Does that come close to answering your question?

**Comment by oubiwann** [comment]
Quick notes on development for version 0.2.0:
 * The tasks listed in #27 will each get their own task branches, to be merged into the branch `feature/27/add-guides` 
 * Those task branches can be anywhere and named anything :-) When a PR is submitted, though, it should not be done against `master` but rather against the task parent branch `feature/27/add-guides` 
 * All docs tickets will be getting merged into `epic/24/add-docs` where they will be tested in the context of the epic before getting merged into `release/0.2.0`, where all epics and their features will get tested together
 * There is a similar structure for the work being done on implementation epic ...
